src/background/controllers/cryptoController.ts

```diff
@@ -64,26 +63,36 @@ export default class CryptoController extends IController {
 
   /*
   * Derives the password hash with the password input.
-  * @return Undefined or error.
   */
-  public derivePasswordHash = async (password: string): Promise<any> => {
-    return new Promise((resolve: any, reject: any) => {
-      setTimeout(() => {
-        try {
-          if (!this.appSalt) {
-            throw Error('appSalt should not be empty');
-          }
+  public derivePasswordHash = async (password: string) => {
+    setTimeout(() => {
+      if (!this.appSalt) {
+        throw Error('appSalt should not be empty');
+      }
+      this.callScryptWebWorker(password, this.appSalt);
```

Comment:
inline the function here

Reply:
do you mean to turn it into a ternary? I I think it is kind of hard to read.  It doesn't seem like you can just throw in a ternary

this.appSalt ? this.callScryptWebWorker(password, this.appSalt) 
        : function() { throw Error('appSalt should not be empty') }()

Reply:
Inlining is moving the entire logic of callScryptWebWorker into the derivePasswordHash func.